themes/src/main/resources/theme/base/admin/resources/js/controllers/clients.js

```diff
@@ -18,11 +18,54 @@ module.controller('ClientTabCtrl', function(Dialog, $scope, Current, Notificatio
     };
 });
 
-module.controller('ClientRoleListCtrl', function($scope, $location, realm, client, roles, $route, RoleById, Notifications, Dialog) {
+module.controller('ClientRoleListCtrl', function($scope, $route, realm, client, ClientRoleList, RoleById, Notifications, Dialog) {
     $scope.realm = realm;
-    $scope.roles = roles;
+    $scope.roles = [];
     $scope.client = client;
 
+    $scope.query = {
+        realm: realm.realm,
+        client: $scope.client.id,
+        search : null,
+        max : 20,
+        first : 0
+    }
+
+    $scope.$watch('query.search', function (newVal, oldVal) {
+        if($scope.query.search && $scope.query.search.length >= 3) {
+            $scope.firstPage();
+        }
+    }, true);
```

Comment:
We should probably be consistent with other search fields in Admin Console. AFAIK, the search always requires pressing enter key and doesn't perform the search dynamically when typing.

Reply:
At the ends, the Client pages role do the same thing, it search without pressing enter key. It's using javascript but for the user and the UI experience it the same so my version is not really incoherent thought, it's going in the good direction for me.